src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copiedStateObject = { ...state };
+  const newState = [];
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
Use a destructuring for action
```suggestion
  for (const action of actions) {
    const {} = action;
   
    switch (type) {
```